crates/asm/manifest-types/src/lib.rs

```diff
@@ -0,0 +1,18 @@
+//! Types for ASM manifest and logging.
+//!
+//! This crate contains the core types used for ASM manifests and log entries,
+//! separated from the main ASM common crate to avoid circular dependencies.
+
+mod errors;
+mod log;
+mod manifest;
+mod payloads;
+
+pub use errors::*;
+pub use log::*;
+pub use manifest::*;
+pub use payloads::*;
+
+/// Type alias for a 32-byte hash.
+// TODO use Buf32 from identifiers?
```

Comment:
Yes, why not? We already have it and it has `ZeroizeOnDrop`.

Reply:
I have no idea, I didn't want to meaningfully change the impl beyond what I strictly had to to make the STF stuff work, just reorganize things, that's what it had iirc